indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordRedirectInfo.java

```diff
@@ -53,15 +53,15 @@ public boolean doLocal(String requestURI)
   }
 
   @Override
-  public URL getRedirectURL(String scheme, String queryString, String requestURI)
+  public URL getRedirectURL(String queryString, String requestURI)
   {
     try {
       final String leader = taskMaster.getCurrentLeader();
       if (leader == null || leader.isEmpty()) {
         return null;
       }
 
-      String location = StringUtils.format("%s://%s%s", scheme, leader, requestURI);
```

Comment:
Can we document somewhere that requests sent using "http" scheme would be redirected to "https" if both TLS/non-TLS ports are enabled?

This might be surprising to a user if the client being used can't handle HTTPS and fails because of the scheme change.

Ideally I think the redirect should follow the same scheme as the original request, but that doesn't seem straightforward to implement since the leaderID always prefers TLS if it's there. Any ideas on whether we could accomplish that in a simple way?

Reply:
yeah, ideally http should get redirected to http but that is not simple enough. however fixing it so that it atleast redirects to a valid url is necessary. so the upgrade path is to enable clients so that they can handle both http and https... then upgrade druid and enabled both http and https ports... then change client configs to refer to druid with https and then disable http port on druid servers.

I'll update the tls specific doc page.

Reply:
updated the doc